addons/mail/static/src/messaging/component/message_list/message_list.js

```diff
@@ -283,85 +276,98 @@ class MessageList extends Component {
     }
 
     /**
-     * Update the scroll position of the message list.
-     * This is not done in patched/mounted hooks because scroll position is
-     * dependent on UI globally. To illustrate, imagine following UI:
-     *
-     * +----------+ < viewport top = scrollable top
-     * | message  |
-     * |   list   |
-     * |          |
-     * +----------+ < scrolltop = viewport bottom = scrollable bottom
-     *
-     * Now if a composer is mounted just below the message list, it is shrinked
-     * and scrolltop is altered as a result:
-     *
-     * +----------+ < viewport top = scrollable top
-     * | message  |
-     * |   list   | < scrolltop = viewport bottom  <-+
-     * |          |                                  |-- dist = composer height
-     * +----------+ < scrollable bottom            <-+
-     * +----------+
-     * | composer |
-     * +----------+
-     *
-     * Because of this, the scroll position must be changed when whole UI
-     * is rendered. To make this simpler, this is done when <Thread/> component
-     * is patched. This is acceptable when <Thread/> has a fixed height,
-     * which is the case for the moment. Also changes to message list that
+     * @returns {mail.messaging.entity.ThreadViewer}
      */
-    updateScroll() {
-        if (this.storeProps.messages.length === 0) {
-            return;
+    get threadViewer() {
+        return this.env.entities.ThreadViewer.get(this.props.threadViewer);
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {Object} hint
+     */
+    async _adjustFromChangeOfThreadCache(hint) {
+        const threadCache = this.threadViewer.threadCache;
+        let isProcessed = false;
+        if (threadCache.isLoaded) {
+            if (threadCache.messages.length > 0) {
+                if (this.threadViewer.threadCacheInitialPosition !== undefined) {
+                    this.el.scrollTop = this.threadViewer.threadCacheInitialPosition;
+                    isProcessed = true;
+                } else {
+                    const lastMessage = threadCache.lastMessage;
+                    if (this.messageRefFromId(lastMessage.id)) {
+                        await this._scrollToLastMessage();
+                        isProcessed = true;
+                    }
+                }
+            } else {
+                isProcessed = true;
+            }
```

Comment:
quick return would be cleaner than many indents (maybe not possible for all level but at least for the top level it is)